core/src/main/scala/org/apache/spark/eventhubs/client/CachedEventHubsReceiver.scala

```diff
@@ -386,7 +386,35 @@ private[spark] object CachedEventHubsReceiver extends CachedReceiver with Loggin
         CachedEventHubsReceiver(ehConf, nAndP, requestSeqNo)
       })
     }
-    receiver.receive(requestSeqNo, batchSize)
+    try {
+      receiver.receive(requestSeqNo, batchSize)
+    } catch {
+      case completionExecution: CompletionException =>
+        val exceptionCause = completionExecution.getCause
+        if (exceptionCause != null &&  exceptionCause.isInstanceOf[RejectedExecutionException] && exceptionCause.getMessage.contains("ReactorDispatcher")) {
+          // reactor dispatcher closed case
+          logInfo(s"(TID $taskId) EventHubsCachedReceiver receive execution for namespaceUri ${ehConf.namespaceUri} " +
+            s"EventHubNameAndPartition $nAndP consumer group ${ehConf.consumerGroup.getOrElse(DefaultConsumerGroup)} " +
+            s"failed with $completionExecution. Try to recerate the entire CachedEventHubsReceiver instance in order to " +
+            s"use a fresh EventHubClient from the underlying java SDK, then try receiving events again.")
+          receiver.client.close();
+          receiver = CachedEventHubsReceiver(ehConf, nAndP, requestSeqNo)
+          receivers.synchronized {
+            receivers.update(key(ehConf, nAndP), receiver)
+          }
+          receiver.receive(requestSeqNo, batchSize)
```

Comment:
What if this attempt also throws? Maybe this change should restructured as a do-while or similar: the loop exits because of a successful receive, the loop exits because we throw out, or we handle the exception and loop so the retry has all the same protections as the first attempt. Might want a retry count and exit on too many retries also so that we don't loop forever if something is bad.

Reply:
This handler just takes care of the case where the ReactorDispatcher has been closed and the client is not usable anymore. If recreating the client doesn't resolve the issue then probably a more serious issue is going on and it's better to fail the job.  Also, this client recreation would be tried up to 4 times by spark, since a failure results in failing the task and spark retries tasks 4 times before failing a job. Therefore, I'm not sure if a do-while would be necessary here.

Reply:
OK, so effectively we already have 4x retry on this, just at a different level. Good enough.